bin/lib/installable/archives.py

```diff
@@ -49,23 +49,27 @@ def __init__(self, install_context: InstallationContext, config: Dict[str, Any])
         compression = self.config_get("compression", "xz")
         if compression == "xz":
             self.s3_path = f"{s3_path_prefix}.tar.xz"
-            self.decompress_flag = "J"
+            decompress_flag = "J"
         elif compression == "gz":
             self.s3_path = f"{s3_path_prefix}.tar.gz"
-            self.decompress_flag = "z"
+            decompress_flag = "z"
         elif compression == "bz2":
             self.s3_path = f"{s3_path_prefix}.tar.bz2"
-            self.decompress_flag = "j"
+            decompress_flag = "j"
         else:
             raise RuntimeError(f"Unknown compression {compression}")
+        self.tar_cmd = ["tar", f"{decompress_flag}xf", "-"]
+        extract_xattrs = self.config_get("extract_xattrs", False)
+        if extract_xattrs:
+            self.tar_cmd += ["--xattrs"]
```

Comment:
Just noting here, there may be aspects of our sandbox or NFS that prevent xattrs from working exactly as you'd like.